splash/browser_tab.py

```diff
@@ -1078,6 +1093,65 @@ def log(self, message, min_level=None):
         log.msg(message, system='render')
 
 
+class ElementsStorage(QObject):
+    def __init__(self, parent):
+        self.name = str(uuid.uuid1())
+        super(ElementsStorage, self).__init__(parent)
+
+    @pyqtSlot(name="getId", result=str)
+    def get_id(self):
+        return str(uuid.uuid1())
+
+
+class HTMLElement(object):
+    DIMENSIONS_JS_FUNC = """
+        (function(element) {
+            var rect = element.getBoundingClientRect();
```

Comment:
As a sidenote, @mike1808, would it be useful to return `nil` from `png` method if the element is not visible?